menpo/base.py

```diff
@@ -210,9 +191,15 @@ def set_target(self, new_target):
         ----------
         new_target : :map:`PointCloud`
             The new target that this object should try and regenerate.
+
+        Returns
+        -------
+        self : :map:`Targetable`
+            A reference to this instance.
         """
         self._target_setter_with_verification(new_target)  # trigger the update
         self._sync_state_from_target()  # and a sync
+        return self
```

Comment:
This seems like quite a large change in behavior, why add it?

On a mutable setter I would argue that it is clearer to **not** return self, that way there is no sense for the programmer that you might be using a immutable setting API.